src/js/jsx/sections/style/Shadow.jsx

```diff
@@ -445,10 +445,19 @@ define(function (require, exports) {
             if (layers.isEmpty()) {
                 return null;
             }
-
-            // Group into arrays of dropShadows, by position in each layer
-            var dropShadowGroups = collection.zip(collection.pluck(layers, "dropShadows")),
-                dropShadowList = dropShadowGroups.map(function (dropShadows, index) {
+                
+            var shadowCountsUnmatch = layers.size > 1 &&
+                    Immutable.Set(layers.map(function (l) { return l.dropShadows.size; })).size !== 1,
```

Comment:
your suggestion looks great! BTW: it still needs the `layers.size > 1 ` to exclude single layer selection.